### ISSUE TITLE
unable to setup anova wifi model a4

### ISSUE THREAD

**Issue by dshokouhi** [Lash-L/anova-wifi-hacs#11]
## Version of the custom_component

e02770c

## Configuration

Installed repo via HACS, then added device ID from anova app. Got the below setup error
```yaml

```

## Describe the bug

Unable to setup the integration

## Debug log

```text

Traceback (most recent call last):
  File "/srv/homeassistant3.9/lib/python3.9/site-packages/homeassistant/config_entries.py", line 382, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/home/homeassistant/.homeassistant/custom_components/anova_sous_vide/__init__.py", line 25, in async_setup_entry
    update = await apc.update(entry.data["device_id"])
  File "/srv/homeassistant3.9/lib/python3.9/site-packages/anova_wifi/parser.py", line 74, in update
    AnovaPrecisionCookerSensor.FIRMWARE_VERSION: anova_status[
KeyError: 'system-info'
```

**Comment by dshokouhi** [comment]
Just a bit of info about my HA core setup. I am running things in a venv and have yet to update python to 3.10. I plan to do so before the next HA update obviously. Hopefully this error isn't due to that but wanted to mention it. I am running HA core 2023.1.7

**Comment by Lash-L** [comment]
Super interesting. Basically it seems like your http request doesn't have some of the same tags mine does.

can you go to https://anovaculinary.io/devices/{your_device_id}/states/?limit=1 and send me the contents of that page?

**Comment by dshokouhi** [comment]
```
[
	{
		"body": {
			"boot-id": "43435004281963",
			"job": {
				"cook-time-seconds": 0,
				"id": "98827197488379",
				"mode": "IDLE",
				"ota-url": "",
				"target-temperature": 65.55,
				"temperature-unit": "F"
			},
			"job-status": {
				"cook-time-remaining": 0,
				"job-start-systick": 1674,
				"provisioning-pairing-code": 0,
				"state": "",
				"state-change-systick": 1674
			},
			"network-info": {
				"bssid": "HIDDEN",
				"connection-status": "connected-station",
				"is-provisioning": false,
				"mac-address": "HIDDEN",
				"mode": "station",
				"security-type": "WPA2",
				"ssid": "HIDDEN"
			},
			"pin-info": {
				"device-safe": 1,
				"water-leak": 0,
				"water-level-critical": 1,
				"water-temp-too-high": 0
			},
			"system-info-3220": {
				"firmware-version": "1.4.4",
				"firmware-version-raw": "VM176_A_01.04.04",
				"largest-free-heap-size": 28008,
				"stack-low-level": 232,
				"stack-low-task": 7,
				"systick": 6154,
				"total-free-heap-size": 28632
			},
			"system-info-nxp": {
				"version-string": ""
			},
			"temperature-info": {
				"heater-temperature": 17.88,
				"triac-temperature": 19.24,
				"water-temperature": 19
			}
		},
		"header": {
			"created-at": "2023-01-29T00:06:16.874554Z",
			"e-tag": "f3948adb476fe680798cad480a6f85a0acc367d1b3b74f5d3c1afa4513823af2",
			"entity-id": "69630009806838"
		}
	}
]
```

**Comment by Lash-L** [comment]
Nice thank you, I should be able to resolve this pretty easily. What model of Sous vide do you have?

**Comment by dshokouhi** [comment]
AN500-US00

the anova wifi 1000w model

**Comment by Lash-L** [comment]
Okay updated! Give it another try. You may need to remove the repository and re-add it maybe even a restart, not fully sure how hacs handles repo updates.

**Comment by dshokouhi** [comment]
ok updated to version 1.0.1, I basically asked HACS to redownload and saw the latest commit. Now I see a slightly different error :)

```
Traceback (most recent call last):
  File "/srv/homeassistant3.9/lib/python3.9/site-packages/homeassistant/config_entries.py", line 382, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/home/homeassistant/.homeassistant/custom_components/anova_sous_vide/__init__.py", line 25, in async_setup_entry
    update = await apc.update(entry.data["device_id"])
  File "/srv/homeassistant3.9/lib/python3.9/site-packages/anova_wifi/parser.py", line 78, in update
    AnovaPrecisionCookerSensor.FIRMWARE_VERSION: anova_status[system_info][
KeyError: 'firmware-version'
```

**Comment by Lash-L** [comment]
oops, forgot you had that 'system-info-nxp' - should be resolved now.

Let's see what bug pops up next :)

**Comment by dshokouhi** [comment]
So after a restart I had a timeout error but it did initialize. I then asked the config entry to reload and I got a device and entities! 

![image](https://user-images.githubusercontent.com/1634145/215298668-31c5b596-62f9-4691-9c10-7d136ee47e0f.png)

I did see them go `unavailable` for about a minute but then they got all the proper data! I did notice that the URL you gave me sometimes would output `[]` then a refresh or 2 later and the data was there.

Thank you!
